### DISCUSSION TITLE
Redirecting the entire page after form is processed on server?

### DISCUSSION THREAD

**Discussion by vladstudio** [unpoly/unpoly#532] [Q&A] [answered]
I might be missing something very obvious. Imagine I have a simple sign in form (not in a modal), like here:

https://unpoly.com/validation

The docs make it clear how to validate the form, but I have no clue what to do next when the form is validated successfully. Suppose its URL is mysite.com/signin - how do I redirect the user to mysite.com/home after server approved the form?

**Comment by vladstudio** [comment]
To clarify: in HTMX, I'd include `HX-Redirect: /home` to server response, is there anything similar for that in Unpoly? 

I tried 

``` 
Header('X-Up-Location: /home');
Header('X-Up-Target: body');
die;
```

and the URL in the addressbar changes to /home, but there is no actual redirect, nothing is loaded.

**Comment by alexandremjacques** [comment, marked as answer]
Your server would do the redirect after validation. 

Unpoly mostly works on server responses. After redirect, the new page is sent to the client (browser) where Unpoly would, then, do it’s magic.

**Reply by vladstudio** [reply]
ooh you mean the usual server redirect, like `header('Location: /home');` ? Nothing unpoly-specific? I tried, and it seems to work. Woo-hoo! Thanks.
